### PR TITLE
added solution for homework Clone

### PR DESCRIPTION
**Pull request by Valdes2**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,45 @@
...
+        this.model = model;
+        this.weight = weight;
+        this.maxSpeed = maxSpeed;
+        this.price = price;
+        this.engine = engine;
+    }
+
+    @Override
+    protected Object clone() throws CloneNotSupportedException {
+        return super.clone();
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Car.java]: @Valdes2 
what if you will do next?

    
public static void main(String[] args) throws CloneNotSupportedException {
        Plane plane = new Plane();
        plane.setName("hello!");
        plane.setEngine(new Engine("type 1"));
        Plane clone = (Plane) plane.clone();
        clone.setName("buy-buy");
        clone.getEngine().setType("type 2");
        System.out.println(plane);
        System.out.println(clone);
    }

Comment (Valdes2) [reply src/main/java/core/basesyntax/Car.java]: fixed

File: src/main/java/core/basesyntax/Car.java [hunk 2]
```diff
@@ -1,5 +1,45 @@
...
+        this.model = model;
+        this.weight = weight;
+        this.maxSpeed = maxSpeed;
+        this.price = price;
+        this.engine = engine;
+    }
+
+    @Override
+    protected Object clone() throws CloneNotSupportedException {
+        return new Car(name, model, weight, maxSpeed, price, engine);
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Car.java]:
```suggestion
        return new Car(name, model, weight, maxSpeed, price, engine.clone());
```

Comment (Valdes2) [reply src/main/java/core/basesyntax/Car.java]: Fixed

**Pull request closed without merging**